src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -669,6 +670,19 @@ public static bool IsRunningFromVisualStudio()
             return BuildEnvironmentHelper.Instance.Mode == BuildEnvironmentMode.VisualStudio;
         }
 
+        public static bool RegisterAnalyzer(string pathToAssembly, LoggingContext loggingContext)
+        {
+            pathToAssembly = FileUtilities.GetFullPathNoThrow(pathToAssembly);
+            if (File.Exists(pathToAssembly))
+            {
+                loggingContext.LogBuildEvent(new BuildCheckAcquisitionEventArgs(pathToAssembly));
+
+                return true;
+            }
+
```

Comment:
Should we log some info in the `false` case? The nonexistent path might be a valuable investigation data point

Reply:
done